src/test/java/com/agilethought/atsceapi/validator/history/HistoryValidatorTest.java

```diff
@@ -0,0 +1,471 @@
+package com.agilethought.atsceapi.validator.history;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import com.agilethought.atsceapi.domain.FixedExpenses;
+import com.agilethought.atsceapi.dto.history.NewHistoryRequest;
+import com.agilethought.atsceapi.exception.BadRequestException;
+import com.agilethought.atsceapi.exception.GlobalExceptionBody.ErrorDetails;
+import com.agilethought.atsceapi.repository.UserRepository;
+
+public class HistoryValidatorTest {
+
+	public NewHistoryRequest newHistoryRequest;
+
+	public FixedExpenses fixedExpenses;
+
+	@Mock
+	public UserRepository userRepository;
+
+	@InjectMocks
+	public HistoryValidator historyValidator;
+
+	@Before
+	public void setUp() {
+		MockitoAnnotations.initMocks(this);
+
+		populateHistoryRequest();
+	}
+
+	public void populateHistoryRequest() {
+		fixedExpenses = new FixedExpenses();
+		fixedExpenses.setRent(1.0);
+		fixedExpenses.setTransport(1.0);
+		fixedExpenses.setInternet(1.0);
+		fixedExpenses.setFeed(1.0);
+		fixedExpenses.setOthers(1.0);
+		fixedExpenses.setTotal(5.0);
+
+		newHistoryRequest = new NewHistoryRequest();
+		newHistoryRequest.setType(1);
+		newHistoryRequest.setUser_id("1");
+		newHistoryRequest.setFixedExpenses(fixedExpenses);
+		newHistoryRequest.setTotalHours(1);
+		newHistoryRequest.setTotalDays(1);
+		newHistoryRequest.setCostDay(1.0);
+		newHistoryRequest.setCostHour(1.0);
+		newHistoryRequest.setProjectCost(1.0);
+		newHistoryRequest.setTaxIVA(1.0);
+		newHistoryRequest.setTaxISR_r(1.0);
+		newHistoryRequest.setTaxIVA_r(1.0);
+		newHistoryRequest.setTotal(1.0);
+		newHistoryRequest.setRevenue(1.0);
+		newHistoryRequest.setStatus(1);
+	}
+
+	@Test
+	public void itShouldThrowErrorMessageHistoryInTypeField() {
+
+		Assertions.assertAll(() -> {
+			newHistoryRequest.setType(null);
+			Exception errorMessageException = Assertions.assertThrows(BadRequestException.class, () -> {
+				historyValidator.validate(newHistoryRequest);
+			});
+			Assertions.assertEquals("One or more fields are invalid", errorMessageException.getMessage());
```

Comment:
You are using a lot times "One or more fields are invalid", convert to constant or use constant defined in constant class